wandb/sdk/wandb_run.py

```diff
@@ -2787,6 +2341,537 @@ def mark_preempting(self) -> None:
         if self._backend and self._backend.interface:
             self._backend.interface.publish_preempting()
 
+    # ------------------------------------------------------------------------------
+    # HEADER
+    # ------------------------------------------------------------------------------
+    @staticmethod
```

Comment:
Can you plz explain why make these static and not "attached" to an instance so that you don't need to pass things like `self._settings` etc.? I'm guessing you expect something to brake with wandb-service, right?

Reply:
`wandb-service` doesn't have access to the run object when we print see here (it only has a stream object):
https://github.com/wandb/client/pull/3193/files#diff-84801d186e1bab60fbbdb1f62605957cc93109cc534d9752297e2177d65d6da9R258

so I need a way to reuse the printing logic (to make it look the same for the single run)
at first I had a shard third file, but it felt awkward, my latest solution is to make all the printing functions static to allow sharing with `service` path. Open to suggestion if you think of a better way to do it :)